cabal-install/src/Distribution/Client/CmdOutdated.hs

```diff
@@ -220,18 +220,23 @@ outdatedAction (ProjectFlags{flagProjectFileName}, OutdatedFlags{..}) _targetStr
   config <- loadConfigOrSandboxConfig verbosity globalFlags
   let globalFlags' = savedGlobalFlags config `mappend` globalFlags
       configFlags  = savedConfigureFlags config
-  (comp, platform, _progdb) <- configCompilerAux' configFlags
   withRepoContext verbosity globalFlags' $ \repoContext -> do
     when (not newFreezeFile && isJust mprojectFile) $
       die' verbosity $
         "--project-file must only be used with --v2-freeze-file."
 
     sourcePkgDb <- IndexUtils.getSourcePackages verbosity repoContext
+    (comp, platform, _progdb) <- configCompilerAux' configFlags
     deps <- if freezeFile
             then depsFromFreezeFile verbosity
             else if newFreezeFile
-                then depsFromNewFreezeFile verbosity mprojectFile
-                else depsFromPkgDesc verbosity comp platform
+                then do
+                       httpTransport <- configureTransport verbosity
```

Comment:
Why is this needed with `newFreezeFile`, but not with `freezeFile`? And how is this related to the main PR's topic?

Reply:
a freeze file is an old style freeze file which is a "~/.cabal.config" file. a new freeze rile is really a project file. so to parse it we handle fully imports and conditionals as well now, which requires a bit more setup.

Reply:
Got it, thanks.